sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/ingestion/models/CustomPropertiesLog.java

```diff
@@ -177,12 +106,65 @@ private static void writePropertyValue(JSONStringer writer, Object value) throws
             JSONUtils.write(writer, PROPERTY_VALUE, value);
         } else if (value instanceof Date) {
             JSONUtils.write(writer, PROPERTY_TYPE, PROPERTY_TYPE_DATETIME);
-            JSONUtils.write(writer, PROPERTY_VALUE, DATETIME_FORMAT.get().format((Date) value));
+            JSONUtils.write(writer, PROPERTY_VALUE, JSONDateUtils.toString((Date) value));
         } else if (value instanceof String) {
             JSONUtils.write(writer, PROPERTY_TYPE, PROPERTY_TYPE_STRING);
             JSONUtils.write(writer, PROPERTY_VALUE, value);
         } else {
             throw new JSONException("Invalid value type");
         }
     }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    /**
+     * Get the properties value.
+     *
+     * @return the properties value
```

Comment:
We started off the generated code from autorest for Java then edited those files, hence they have a different style (the json ones) from other files.

Will refactor every JSON file at once someday but this PR is out of scope.